compiler/lexer/lexer.go

```diff
@@ -389,6 +399,69 @@ func (l *Lexer) peekChar() rune {
 	// Peek shouldn't increment positions.
 }
 
+func (l *Lexer) reducePositiveSigns() {
+	if isPositiveSign(l.ch) {
+		if l.isUnary() {
+			if isPositiveSign(l.peekChar()) {
+				l.readChar()
+				for isPositiveSign(l.peekChar()) || isIdentifier(l.peekChar()) || isNegativeSign(l.peekChar()) {
+					l.readChar()
+				}
+			} else if isIdentifier(l.peekChar()) {
+				l.readChar()
+			}
+		}
+	}
+}
+
+func (l *Lexer) skipUnary() {
+	if l.ch == '+' && l.isUnary() {
+		l.readChar()
+	}
+}
+
+func (l *Lexer) isUnary() bool {
```

Comment:
I think this should be a helper function that receives an `rune` argument just like `isIdentifier`, instead of a `lexer`'s function. We should change either its behavior or its name for the consistency.

Reply:
First I thought as that. But this helper function goes backward the lexer so `*Lexer` is needed.
Passing `*Lexer` via argument might be ineffective.
I will change the function name to `checkUnary()`.